plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -140,6 +155,51 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
     return associated_units, failed_units
 
 
+def get_recursive_module_deps(module_units):
+    """
+    Module Metadata specifies dependencies on other modules. This method is useful
+    for taking a set of Modulemd documents and returning the total set of module
+    dependencies for those units.
+
+    :param module_units: iterable of Modulemd
+    :type module_units: iterable
+
+    :return: set of Modulemd
+    :rtype: set
+    """
+    module_deps = set()
+    for unit in module_units:
+        for dependency in unit.dependencies:
+            # we're not a real client, so we don't care about the platform
+            # we need to copy all dependencies for any platform
+            dependency.pop("platform")
+
+            for name, streams in dependency.items():
+                all_module_streams = models.Modulemd.objects.filter(name=name)
```

Comment:
Streams like "master" and "rolling"  will get new versions on a regular basis, and if those don't get cleaned out regularly with force-full syncs, then we're going to be copying (and thus doing all of these recursive operations on) a lot of of module documents that will likely never be used. 

I don't know if that's actually a problem, I just want to point it out.

Reply:
I think the ask from Katello was to copy all versions but we can ask again if that's still the case.
I wonder if the version for different architectures and context is/can be the same and if version bump means that some artifact changed or some other metadata.
IMO, copy of top-level modules and dependent ones should have the same logic.

Reply:
The only documentation I've found is this, but it's relatively informative

https://fedoraproject.org/wiki/Infrastructure/Factory2/Focus/StreamExpansion#New_Problems

Reply:
Hmmm, from those docs, it makes sense to copy the latest version of a dependency but all contexts.